repo/repo.go

```diff
@@ -25,6 +25,8 @@ type Repo interface {
 	// SetAPIAddr sets the API address in the repo.
 	SetAPIAddr(addr string) error
 
+	Self() Repo
```

Comment:
I needed a way to be able to get to the `FSRepo` and hence get to the `filestore.Datastore` in the next commit (4ef5531).  In this commit `node.Repo.(*fsrepo.FSRepo)` does not work as node.Repo is `repo.ref` so I needed to use add the Self() method and use `node.Repo.Self().(*fsrepo.FSRepo)`